test/approvedTransfer.js

```diff
@@ -89,372 +92,213 @@ describe("Test Approved Transfer", function () {
-      const success = parseRelayReceipt(txReceipt);
-      assert.isNotOk(success, "transfer should fail with 1 guardian confirmation");
-      // should succeed with 2 confirmations
-      await manager.relay(transferModule, "transferToken",
-        [wallet.contractAddress, ETH_TOKEN, recipient.address, amountToTransfer, ZERO_BYTES32], wallet,
-        [owner, ...sortWalletByAddress([guardian1, guardian2])]);
-      const after = await deployer.provider.getBalance(recipient.address);
-      assert.equal(after.sub(before).toNumber(), amountToTransfer, "should have transfered the ETH amount");
-    });
-    it("should fail to transfer ETH when signer is not a guardians", async () => {
-      const amountToTransfer = 10000;
-      const count = (await guardianManager.guardianCount(wallet.contractAddress)).toNumber();
-      assert.equal(count, 0, "0 guardians should be active");
-      // should fail
-      const txReceipt = await manager.relay(transferModule, "transferToken",
-        [wallet.contractAddress, ETH_TOKEN, recipient.address, amountToTransfer, ZERO_BYTES32], wallet, [owner, guardian1]);
-      const success = parseRelayReceipt(txReceipt);
-      assert.isNotOk(success, "transfer should fail when signer is not a guardian");
-    });
-    it("should transfer ERC20 with 1 confirmations for 1 guardians", async () => {
-      const amountToTransfer = 10000;
-      await addGuardians([guardian1]);
-      const count = (await guardianManager.guardianCount(wallet.contractAddress)).toNumber();
-      assert.equal(count, 1, "1 guardians should be active");
-      const before = await erc20.balanceOf(recipient.address);
-      // should succeed with one confirmation
-      await manager.relay(transferModule, "transferToken",
-        [wallet.contractAddress, erc20.contractAddress, recipient.address, amountToTransfer, ZERO_BYTES32], wallet, [owner, guardian1]);
-      const after = await erc20.balanceOf(recipient.address);
-      assert.equal(after.sub(before).toNumber(), amountToTransfer, "should have transfered the ERC20 amount");
-    });
-    it("should only transfer ERC20 with 2 confirmations for 3 guardians", async () => {
-      const amountToTransfer = 10000;
-      await addGuardians([guardian1, guardian2, guardian3]);
-      const count = (await guardianManager.guardianCount(wallet.contractAddress)).toNumber();
-      assert.equal(count, 3, "3 guardians should be active");
-      const before = await erc20.balanceOf(recipient.address);
-      // should fail with one confirmation
-      const txReceipt = await manager.relay(transferModule, "transferToken",
-        [wallet.contractAddress, erc20.contractAddress, recipient.address, amountToTransfer, ZERO_BYTES32], wallet, [owner, guardian1]);
-      const success = parseRelayReceipt(txReceipt);
-      assert.isNotOk(success, "transfer with 1 guardian signature should fail");
-      // should succeed with 2 confirmations
```

Comment:
My goal was to address your comment above (re: bundling multiple tests into one & repeating success/failure checks) as well as reducing the amount of code duplication. I also tried to make the tests shorter and easier to read. 

Regarding the nested `describe`s, I generally find it less clear to have 8 tests on the same level of indentation (you have to read every single test description to understand what is covered or not in these 8 tests) rather than grouping tests by e.g. the number of guardians used.

I would agree that the `describe` "Approved by 1 EOA and 2 smart-contract guardians" under "Contract Call" and "Approve token and Contract call" may be removed, but I kept them for consistency and to highlight the fact that for "Transfer" we tested 3 different combination of EOA and smart-contract guardians, whereas for "Contract Call" and "Approve token and Contract call" we only used one such combination.